src/OnboardingSPA/components/Drawer/DrawerPanel/DesignColors.js

```diff
@@ -1,15 +1,378 @@
+						break;
+				}
+			}
+
+			selectedGlobalStyle.settings.color.palette.theme = selectedThemeColorPalette;
+			setGlobalStyles(selectedGlobalStyle);
+			updatePreviewSettings(
+				useGlobalStylesOutput(selectedGlobalStyle, storedPreviewSettings)
+			);
+
+			return selectedGlobalStyle;
+		}
+	}
+
+	const getColorStylesAndPatterns = async () => {
+		const globalStyles = await getGlobalStyles();
+		let selectedGlobalStyle;
+		if (currentData?.data?.theme?.variation) {
+			selectedGlobalStyle = globalStyles.body.filter(
+				(globalStyle) =>
+					globalStyle.title === currentData.data.theme.variation
+			)[0];
+		} else {
+			selectedGlobalStyle = globalStyles.body[0];
+		}
+		setGlobalStyles(selectedGlobalStyle);
+
+		let selectedColors;
+		if (!currentData?.data?.palette[0]?.hasOwnProperty('supports')) {
+			currentData.data.palette[0] = {
+				"slug": "",
+       			"name": "",
+       			"color": [
+					{"slug": "primary", "name": "Primary", "color": ""},
+					{ "slug": "secondary", "name": "Secondary", "color": ""},
+					{ "slug": "tertiary", "name": "Tertiary", "color": ""},
+					{ "slug": "background", "name": "Background", "color": ""},
+				],
+				"supports": ["yith-wonder"]
+			};
+			selectedColors = currentData.data.palette[0];
+			setCurrentOnboardingData(currentData);
+		}
+		else {
+			selectedColors = currentData.data.palette[0];
+
+			if(selectedColors.slug === 'custom') {
+				setBackgroundColor(selectedColors?.color[3].color ?? null);
+				setPrimaryColor(selectedColors?.color[0].color ?? null);
+				setSecondaryColor(selectedColors?.color[1].color ?? null);
+				setTertiaryColor(selectedColors?.color[2].color ?? null);
+			}
+		} 
+		setSelectedColors(selectedColors);
+		setThemeColorPalette(currentData?.data?.palette[0]['slug'], selectedColors, selectedGlobalStyle);
+		setIsLoaded(true);
+
+	};
+
+	useEffect(() => {
+		if (!isLoaded) getColorStylesAndPatterns();
+	}, [isLoaded]);
+
+	async function setCustomColors() {
+		let selectedGlobalStyle = globalStyles;
+		let selectedThemeColorPalette = selectedGlobalStyle?.settings?.color?.palette?.theme;
+
+		if (selectedThemeColorPalette) {
+			for (let idx = 0; idx < selectedThemeColorPalette.length; idx++) {
+				switch (selectedThemeColorPalette[idx]?.slug) {
+					case 'background':
+						if (colorPickerCalledBy == 'background' && backgroundColor)
+							selectedThemeColorPalette[idx].color = backgroundColor;
+						break;
+					case 'primary':
+						if (colorPickerCalledBy == 'primary' && primaryColor)
+							selectedThemeColorPalette[idx].color = primaryColor;
+						break;
+					case 'secondary':
+						if (colorPickerCalledBy == 'secondary' && secondaryColor)
+							selectedThemeColorPalette[idx].color = secondaryColor;
+						break;
+					case 'tertiary':
+						if (colorPickerCalledBy == 'tertiary' && tertiaryColor)
+							selectedThemeColorPalette[idx].color = tertiaryColor;
+						break;
+				}
+			}
```

Comment:
We can use a better data structure that can eliminate these loops and idx variables